src/util.ts

```diff
@@ -643,3 +645,70 @@ export function intHash(str : string, length = 10) : number {
 
+/**
+ * Evaluate a JSON query expression against the provided data.
+ * @param data The data to evaluate the JSON query against.
+ * @param jsonPath The JSON path or custom JSON query expression.
+ * @param jsonPathOperator The operator to use for comparison.
+ * @param expectedValue The expected value to compare against.
+ * @returns An object containing the status and the evaluation result.
+ * @throws Error if the evaluation returns undefined.
+ */
+export async function evaluateJsonQuery(data: any, jsonPath: string, jsonPathOperator: string, expectedValue: any): Promise<{ status: boolean; response: any }> {
+    // Attempt to parse data as JSON; if unsuccessful, handle based on data type.
+    let response: any;
+    try {
+        response = JSON.parse(data);
+    } catch {
+        response = typeof data === "number" || typeof data === "object" ? data : data.toString();
+    }
+
+    try {
+        // If a JSON path is provided, pre-evaluate the data using it.
+        response = (jsonPath) ? await jsonata(jsonPath).evaluate(response) : response;
+
+        if (typeof response === "object" || Array.isArray(response) || response instanceof Date || typeof response === "function") {
+            throw new Error(`The post-JSON query evaluated response from the server is of type ${typeof response}, which cannot be directly compared to the expected value`);
+        }
+
+        // Perform the comparison logic using the chosen operator
+        let jsonQueryExpression;
+        switch (jsonPathOperator) {
+            case ">":
+            case ">=":
+            case "<":
+            case "<=":
+                jsonQueryExpression = `$number($.value) ${jsonPathOperator} $number($.expected)`;
+                break;
+            case "!=":
+                jsonQueryExpression = "$.value != $.expected";
+                break;
+            case "==":
+                jsonQueryExpression = "$.value = $.expected";
+                break;
+            case "contains":
+                jsonQueryExpression = "$contains($.value, $.expected)";
+                break;
+            default:
+                throw new Error(`Invalid condition ${jsonPathOperator}`);
+        }
+
+        // Evaluate the JSON Query Expression
+        const expression = jsonata(jsonQueryExpression);
+        const status = await expression.evaluate({
+            value: response.toString(),
+            expected: expectedValue.toString()
+        });
+
+        if (response === undefined || status === undefined) {
+            throw new Error("Query evaluation returned undefined. Check query syntax and the structure of the response data");
+        }
+
+        return {
+            status,  // The evaluation of the json query
+            response // The response from the server or result from initial json-query evaluation
+        };
+    } catch (err: any) {
+        throw new Error(`Error evaluating JSON query: ${err.message}. Response from server was: ${response}`);
```

Comment:
response in this context may quite large.
If we just print it to the console, this could DOS the server => please at most print a truncated version.

Alternatively, just returning the error is likely fine too.

Reply:
I noticed the potential for this issue in my own testing and wrote truncation code that was a bit clunky. Let me adjust and see what I can come up with.

Reply:
You can use the same as here https://github.com/louislam/uptime-kuma/pull/4806#discussion_r1625080954



```suggestion
        if (s.length > 50) {
            response = `${s.substring(0, 50)}…`;
        }
        throw new Error(`Error evaluating JSON query: ${err.message}. Response from server was: ${response}`);
```

Reply:
If we implement this truncation at the error case, there are still situations where the `bean.message` can be very long. Is this acceptable, or should we cap the maximum length of JSON responses, globally? For example:
<img width="765" alt="image" src="https://github.com/louislam/uptime-kuma/assets/9312603/56785662-db86-45f0-a24c-1cf609894d50">

Reply:
> so simply removing response from the `bean.message` isn't ideal either

Yes, ideally those would be written to disk and then lasy loaded if the user requests them.
For now, just truncating everything is fine. (this is an issue which affects all monitors, not just this one)

Reply:
Resolved with [9820f57](https://github.com/louislam/uptime-kuma/pull/4717/commits/9820f57c64bc5d8972219fd27d4c6f0aac153855)